### PR TITLE
Fix broken links in documentation

### PR DESCRIPTION
**Pull request by sfauvel**
## Content

This PR fixes some broken links.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [ ] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [x] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

### CODE REVIEW

File: docs/website/root/mithril/mithril-network/aggregator.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ In its initial version, the Mithril aggregator comprises two main components:
 
 :::tip
 
-The Mithril aggregator's **REST API** documentation can be found [here](/aggregator-api).
+The Mithril aggregator's **REST API** documentation can be found [here](/openapi-ui).
```
Comment (jpraynaud) [inline docs/website/root/mithril/mithril-network/aggregator.md:94]: This is probably the correct link :slightly_smiling_face: 
```suggestion
The Mithril aggregator's **REST API** documentation can be found [here](/doc/aggregator-api).
```

**Comment by sfauvel** [comment]
I'm not sure about some changes.
* Is `unstable release` the new `nightly build` ? 
* Is `openapi-ui` the new `aggregator-api`? 

There is still an issue with mithril-client rust doc that is not correctly generated. The reason is that mithril-client and mithril-client-lib has the same output filename m̀ihtril-client`. We can see a warning in logs when generating documentation.

**Comment by jpraynaud** [comment]
> I'm not sure about some changes.
> 
> * Is `unstable release` the new `nightly build` ?

Yes, `nightly` is now `unstable`

> * Is `openapi-ui` the new `aggregator-api`?

There are both `/openapi-ui` and `/doc/aggregator-api` (which links to the former)

> There is still an issue with mithril-client rust doc that is not correctly generated. The reason is that mithril-client and mithril-client-lib has the same output filename m̀ihtril-client`. We can see a warning in logs when generating documentation.

We're planning to rename the output filename to the name of the folder: `mithril-client-cli` would output `mithril-clinet-cli` instead of `mithril-client`. That's another reason to do it :+1:

File: docs/website/versioned_docs/version-maintained/manual/developer-docs/nodes/mithril-client-library.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ It is responsible for handling the different types of data certified by Mithril,
 
 | Node | Source repository | Rust documentation |
 |:-:|:-----------------:|:------------------:|
-**Mithril client** | [:arrow_upper_right:](https://github.com/input-output-hk/mithril/tree/main/mithril-client) | [:arrow_upper_right:](https://mithril.network/mithril-client/doc/mithril_client/index.html) |
+**Mithril client** | [:arrow_upper_right:](https://github.com/input-output-hk/mithril/tree/main/mithril-client) | [:arrow_upper_right:](https://mithril.network/rust-doc/mithril-client/mithril_client/index.html) |
```
Comment (dlachaumepalo) [inline docs/website/versioned_docs/version-maintained/manual/developer-docs/nodes/mithril-client-library.md:41]: Correct link:
```suggestion
**Mithril client** | [:arrow_upper_right:](https://github.com/input-output-hk/mithril/tree/main/mithril-client) | [:arrow_upper_right:](https://mithril.network/rust-doc/mithril_client/index.html) |
```

File: docs/website/root/manual/developer-docs/nodes/mithril-client-library.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ It is responsible for handling the different types of data certified by Mithril,
 
 | Node | Source repository | Rust documentation |
 |:-:|:-----------------:|:------------------:|
-**Mithril client** | [:arrow_upper_right:](https://github.com/input-output-hk/mithril/tree/main/mithril-client) | [:arrow_upper_right:](https://mithril.network/mithril-client/doc/mithril_client/index.html) |
+**Mithril client** | [:arrow_upper_right:](https://github.com/input-output-hk/mithril/tree/main/mithril-client) | [:arrow_upper_right:](https://mithril.network/rust-doc/mithril-client/mithril_client/index.html) |
```
Comment (dlachaumepalo) [inline docs/website/root/manual/developer-docs/nodes/mithril-client-library.md:41]:
```suggestion
**Mithril client** | [:arrow_upper_right:](https://github.com/input-output-hk/mithril/tree/main/mithril-client) | [:arrow_upper_right:](https://mithril.network/rust-doc/mithril_client/index.html) |
```

**Review by ghubertpalo — APPROVED** [review]

**Pull request merged**